core/src/test/scala/org/broadinstitute/dsde/rawls/webservice/SnapshotApiServiceSpec.scala

```diff
@@ -0,0 +1,107 @@
+    try {
+      testCode(apiService)
+    } finally {
+      apiService.cleanupSupervisor
+    }
+  }
+
+  def withTestDataApiServices[T](testCode: TestApiService => T): T = {
+    withDefaultTestDatabase { dataSource: SlickDataSource =>
+      withApiServices(dataSource)(testCode)
+    }
+  }
+
+  def withEmptyTestDataApiServices[T](testCode: TestApiService => T): T = {
+    withEmptyTestDatabase { dataSource: SlickDataSource =>
+      withApiServices(dataSource)(testCode)
+    }
+  }
+
+  it should "return 201 when creating a reference to a snapshot" in withTestDataApiServices { services =>
+    Post(s"${testData.wsName.path}/snapshots", httpJson(
+      DataRepoSnapshot(
+        name = "foo",
+        snapshotId = "realsnapshot"
+      )
+    )) ~>
+      sealRoute(services.snapshotRoutes) ~>
+      check { assertResult(StatusCodes.Created) {status} }
+  }
+
+  it should "return 404 when creating a reference to a snapshot that doesn't exist" in withTestDataApiServices { services =>
+    Post(s"${testData.wsName.path}/snapshots", httpJson(
+      DataRepoSnapshot(
+        name = "foo",
+        snapshotId = "fakesnapshot"
+      )
+    )) ~>
+      sealRoute(services.snapshotRoutes) ~>
+      check { assertResult(StatusCodes.NotFound) {status} }
+  }
+
+  it should "return 404 when creating a reference to a snapshot in a workspace that doesn't exist" in withTestDataApiServices { services =>
+    Post(s"/workspaces/foo/bar/snapshots", httpJson(
+      DataRepoSnapshot(
+        name = "foo",
+        snapshotId = "bar"
+      )
+    )) ~>
+      sealRoute(services.snapshotRoutes) ~>
+      check { assertResult(StatusCodes.NotFound) {status} }
+  }
+
+  it should "return 200 when getting a reference to a snapshot" in withTestDataApiServices { services =>
+    Post(s"${testData.wsName.path}/snapshots", httpJson(
+      DataRepoSnapshot(
+        name = "foo",
+        snapshotId = "realsnapshot"
+      )
+    )) ~>
+      sealRoute(services.snapshotRoutes) ~>
+      check {
+        val response = responseAs[DataRepoSnapshotReference]
+        assertResult(StatusCodes.Created) {status}
+
+        Get(s"${testData.wsName.path}/snapshots/${response.referenceId}") ~>
+          sealRoute(services.snapshotRoutes) ~>
+          check { assertResult(StatusCodes.OK) {status} }
+      }
+  }
+
+  it should "return 404 when getting a reference to a snapshot that doesn't exist" in withTestDataApiServices { services =>
+    Get(s"${testData.wsName.path}/snapshots/${UUID.randomUUID().toString}") ~>
+      sealRoute(services.snapshotRoutes) ~>
+      check { assertResult(StatusCodes.NotFound) {status} }
+  }
+
+  it should "return 404 when getting a reference to a snapshot in a workspace that doesn't exist" in withTestDataApiServices { services =>
+    Get(s"/workspaces/foo/bar/snapshots/doesntmatter") ~>
+      sealRoute(services.snapshotRoutes) ~>
+      check { assertResult(StatusCodes.NotFound) {status} }
+  }
+
```

Comment:
Can you add tests for adding a reference to a workspace that the user doesn't have access to?